pkg/kubelet/apis/config/validation/validation.go

```diff
@@ -142,19 +142,32 @@ func ValidateKubeletConfiguration(kc *kubeletconfig.KubeletConfiguration) error
 	}
 
 	if localFeatureGate.Enabled(features.GracefulNodeShutdown) {
-		if kc.ShutdownGracePeriod.Duration < 0 || kc.ShutdownGracePeriodCriticalPods.Duration < 0 || kc.ShutdownGracePeriodCriticalPods.Duration > kc.ShutdownGracePeriod.Duration {
-			allErrors = append(allErrors, fmt.Errorf("invalid configuration: ShutdownGracePeriod %v must be >= 0, ShutdownGracePeriodCriticalPods %v must be >= 0, and ShutdownGracePeriodCriticalPods %v must be <= ShutdownGracePeriod %v", kc.ShutdownGracePeriod, kc.ShutdownGracePeriodCriticalPods, kc.ShutdownGracePeriodCriticalPods, kc.ShutdownGracePeriod))
+		if localFeatureGate.Enabled(features.PodPriorityBasedGracefulShutdown) {
+			if kc.ShutdownGracePeriod.Duration > 0 || kc.ShutdownGracePeriodCriticalPods.Duration > 0 {
+				allErrors = append(allErrors, fmt.Errorf("invalid configuration: Specifying ShutdownGracePeriod or ShutdownGracePeriodCriticalPods requires feature gate PodPriorityBasedGracefulShutdown disabled"))
```

Comment:
I'm not sure if this is behavior we want. For example, consider case when someone has kubelet config setup with `ShutdownGracePeriod` and `ShutdownGracePeriodCriticalPods`. Then, in future, `PodPriorityBasedGracefulShutdown` feature gate is enabled by default. This will then result in their kubelet config not being valid anymore.

I think alternatively we should relax this and ensure that either `ShutdownGracePeriod` and `ShutdownGracePeriodCriticalPods` are specified OR `PodPriorityShutdownGracePeriods` config is specified.